File: API/Entities/PersonSkill.cs
using System;
using System.ComponentModel.DataAnnotations.Schema;

namespace Skillustrator.Api.Entities
{
    public class PersonSkill : EntityBase
    {
        public int PersonId { get; set; }

        public Person Person { get; set; }

        public int SkillId { get; set; }

        public Skill Skill { get; set; }

        public SkillLevel SkillLevel { get; set; }

        public TimePeriod TimeUsed { get; set; }

        public TimePeriod TimeSinceUsed { get; set; }
    }
}

File: UI/src/app/models/person-skill.ts
import { Person } from './person';
import { Skill } from './skill';
import { SkillLevel } from './skill-level';
import { TimePeriod } from './time-period';

export class PersonSkill {
  id: number;
  person: Person;
  skill: Skill;
  skillLevel: SkillLevel;
  timeUsed: TimePeriod;
  timeSinceUsed: TimePeriod;

  constructor(id: number, person: Person, skill: Skill, skillLevel: SkillLevel, timeUsed: TimePeriod, timeSinceUsed: TimePeriod) {
    this.id = id;
    this.person = person;
    this.skill = skill;
    this.skillLevel = skillLevel;
    this.timeUsed = timeUsed;
    this.timeSinceUsed = timeSinceUsed;
  }
}
